pkg/reconciler/security/istio/httppolicybinding/httppolicybinding_test.go

```diff
@@ -0,0 +1,686 @@
+				withPolicyBindingPolicy(testPolicyName),
+				withPolicyBindingStatusInit(),
+				withPolicyBindingStatusClassCompatible(),
+				withPolicyBindingStatusFailure("AuthorizationPolicyReconcileFailure", `Failed to update Istio AuthorizationPolicy: inducing failure for update authorizationpolicies`),
+			),
+		}},
+		WantEvents: []string{
+			Eventf(corev1.EventTypeWarning, "InternalError", `Failed to update Istio AuthorizationPolicy: inducing failure for update authorizationpolicies`),
+		},
+		WantErr: true,
+	}, {
+		Name: "success",
+		Objects: []runtime.Object{
+			newPolicyBinding(
+				testBindingName, testNamespace,
+				withPolicyBindingSubject("subject"),
+				withPolicyBindingPolicy(testPolicyName),
+			),
+			newSubject("subject", testNamespace,
+				withSubjectAnnotations(map[string]interface{}{"security.knative.dev/authorizableOn": `{"matchLabels":{"app":"test"}}`}),
+			),
+			newPolicy(testPolicyName, testNamespace,
+				withTestPolicySpec(),
+			),
+		},
+		Key: testNamespace + "/" + testBindingName,
+		WantCreates: []runtime.Object{
+			newRequestAuthentication(testBindingName, testBindingName, testNamespace,
+				withRequestAuthenticationTestSpec(),
+			),
+			newAuthorizationPolicy(testBindingName, testBindingName, testNamespace,
+				withAuthorizationPolicyTestSpec(),
+			),
+		},
+		WantStatusUpdates: []clientgotesting.UpdateActionImpl{{
+			Object: newPolicyBinding(testBindingName, testNamespace,
+				withPolicyBindingSubject("subject"),
+				withPolicyBindingPolicy(testPolicyName),
+				withPolicyBindingStatusReady(),
+			),
+		}},
+		WantEvents: []string{
+			Eventf(corev1.EventTypeNormal, "HTTPPolicyBindingReconciled", `HTTPPolicyBinding reconciled: "testnamespace/testpolicybinding"`),
+		},
+	}}
+
+	defer logtesting.ClearAll()
+	table.Test(t, MakeFactory(func(ctx context.Context, listers *Listers, cmw configmap.Watcher, testData map[string]interface{}) controller.Reconciler {
+		ctx = resource.WithDuck(ctx)
+		r := &Reconciler{
+			Base:            reconciler.NewBase(ctx, controllerAgentName, cmw),
+			policyLister:    listers.GetHTTPPolicyLister(),
+			authnLister:     listers.GetRequestAuthenticationLister(),
+			authzLister:     listers.GetAuthorizationPolicyLister(),
+			istioClient:     istioclient.Get(ctx),
+			policyTracker:   tracker.New(func(types.NamespacedName) {}, 0),
+			subjectResolver: security.NewSubjectResolver(ctx, func(types.NamespacedName) {}),
+		}
+
+		return bindingreconciler.NewReconciler(ctx, r.Logger, r.RunClientSet, listers.GetHTTPPolicyBindingLister(), r.Recorder, r)
+	}))
+}
+
+type bindingOption func(*v1alpha1.HTTPPolicyBinding)
+type subjectOption func(*unstructured.Unstructured)
+type policyOption func(*v1alpha1.HTTPPolicy)
+type requestAuthnOption func(*istiosecurityclient.RequestAuthentication)
+type authzPolicyOption func(*istiosecurityclient.AuthorizationPolicy)
+
+func newPolicyBinding(name, namespace string, opts ...bindingOption) *v1alpha1.HTTPPolicyBinding {
```

Comment:
I think I'd rather put all of these guys under `reconciler/testing/` and make them public.
Any particular reason why you want them here?

Reply:
I was trying avoiding messing up existing code. But maybe it's not a big problem because they will end up in different files.